### PR TITLE
feat: gallery item page view counter

### PR DESCRIPTION
**Pull request by newraina**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 \_\_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Parts of #5547 
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer?target=1CAv6Zq3yVxL3eKhC94GWTWVwp1w4jZbqeZ6wXx1rPAhrce&usdamount=0&donation=true)

#### Community participation

- [x] [Are you at KodaDot Ecosystem Telegram?](https://t.me/kodadot_eco)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

CleanShot 2023-12-05-mTucn29P@2x

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 4df3a8c</samp>

Added a page view count feature to the gallery item component using a new composable function and a counter API. Updated the icons plugin and the utils file with the necessary dependencies and constants.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 4df3a8c</samp>

> _`usePageViews` sends_
> _route path to counter API_
> _shows eye icon, count_

### CODE REVIEW

File: components/gallery/GalleryItem.vue [hunk 1]
```diff
@@ -102,10 +102,18 @@
...
+              <span v-if="pageViewCount === null">--</span>
+              <span v-else>{{ formatNumber(pageViewCount) }}</span>
+            </div>
+
+            <div class="is-flex is-flex-direction-row is-flex-wrap-wrap">
               <IdentityItem
                 v-if="nft?.issuer"
-                class="gallery-avatar mr-4"
+                class="gallery-avatar"
+                :class="{ 'mr-4': isMobile, 'mr-8': !isMobile }"
```
Comment (roiLeo) [inline components/gallery/GalleryItem.vue:116]: better?
```suggestion
                :class="isMobile ? 'mr-4' : 'mr-8'"
```

**Review by roiLeo — APPROVED** [review]
every refresh counter is increased maybe it need to check User Agent on worker side

otherwise lgtm ✅

**Review by preschian — APPROVED** [review]

**Comment by exezbcz** [comment]
it does not include retrospective views right


otherwise works nicely 

- could you please match the color in light mode to neutral-7, which is the same in dark mode 
image

image

**Comment by newraina** [comment]
@exezbcz 

> it does not include retrospective views right

Yes.

> could you please match the color in light mode to neutral-7

Done.

To achieve the above, I have updated the neutral color palette in the tailwind config as mentioned in #8137. @roiLeo you may want to review it again. Given that the neutral colors stay the same in both light and dark modes, I didn't use CSS variables in the tailwind config.

File: libs/ui/tailwind.config.js [hunk 2]
```diff
@@ -50,6 +50,22 @@ module.exports = {
...
+          5: '#d9d9d9',
+          6: '#bfbfbf',
+          7: '#8c8c8c',
+          8: '#595959',
+          9: '#454545',
+          10: '#262626',
+          11: '#1f1f1f',
+          12: '#141414',
+          13: '#000000',
+        },
```
Comment (roiLeo) [inline libs/ui/tailwind.config.js:68]: allright, we might ping this when needed.
weird that you can use this tailwind colors outside package

**Review by roiLeo — APPROVED** [review]

File: libs/ui/tailwind.config.js (hunk 2 above)
Comment (newraina) [reply libs/ui/tailwind.config.js:68]: Tailwind config is shared both inside and outside the lib:

https://github.com/kodadot/nft-gallery/blob/47d18dd2136f2b86caa519c932c1a2211ff95d28/nuxt.config.ts#L17

Comment (roiLeo) [reply libs/ui/tailwind.config.js:68]: oh that's why, noted!

**Comment by yangwao** [comment]
Thanks!
pay 30 usd

**Pull request merged**